### PR TITLE
Fix filter for architecture

### PR DESCRIPTION
**Pull request by Martchus**

### CODE REVIEW

**Comment by Martchus** [comment]
I extended this to cover [the other filter-related issue](https://progress.opensuse.org/issues/43973) as well.

**Comment by coolo** [comment]
+21% is not a bad improvement

**Comment by Martchus** [comment]
So we're good to merge? (I pushed more changes after @kraih approved the PR.)

File: lib/OpenQA/Utils.pm [hunk 1]
```diff
@@ -918,8 +918,14 @@ sub compose_job_overview_search_args {
 
     # add simple query params to search args
     for my $arg (qw(distri version flavor build)) {
-        my $param = $controller->param($arg) or next;
-        $search_args{$arg} = $param;
+        my $params = $controller->every_param($arg) or next;
```
Comment (kraih) [inline lib/OpenQA/Utils.pm]: Not a blocker at all, just a style question. Wouldn't this make more sense written as `next unless my $params = ...;`? Sure a variable is created here, but the fact that there is also control flow logic on this line is much more important for understanding the code. And i think that would be more obvious that way.

Comment (Martchus) [reply lib/OpenQA/Utils.pm]: I didn't know that this is possible. I'll try it the next time.

**Pull request merged**